File: Demo/Demo/Models/Engine.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Demo.Models
{
    public enum EngineType { Petrol, Diesel, Electric, Hybrid }

    public class Engine
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Power { get; set; }
        public EngineType Type { get; set; }
        public int Cost { get; set; }

        public int BrandId { get; set; }
        public Brand Brand { get; set; }

        public List<EngineOfModel> EngineOfModels { get; set; }
    }
}

File: Demo/Demo/ClientApp/AutoConfigurator/models/engine.ts
export enum EngineType { Бензиновый, Дизельный, Электрический, Гибридный }

export class Engine {
    constructor(public id?: number, public name?: string, public power?: number, public type?: EngineType, public cost?: number, public brandId?: number) {
        this.cost = 0;
    }
}